spot-ml/src/main/scala/org/apache/spot/utilities/transformation/PrecisionUtility.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spot.utilities.transformation
+
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions._
+
+/**
+  * PrecisionUtility will transform a number from Double to Float if precision option is set to 32 bit,
+  * if default or 64 bit is selected, it will just return the same number type Double.
+  *
+  * This abstract class permits the execution of a single path during the entire analysis. Instead of checking what
+  * type should the data be converted, each implementation will know what type should be converted into and converted
+  * back.
+  *
+  */
+
+sealed trait PrecisionUtility extends Serializable {
+
+  type TargetType
+
+  /**
+    * Converts a number into the precision type; it can be Float (32) or Double (64).
+    * For the Double implementation it will just return the same value without any transformation.
+    *
+    * @param double a number to convert from Double to Target type.
+    * @return
+    */
+  def toTargetType(double: Double): TargetType
+
+  /**
+    * Converts back an Iterable of numbers from TargetType to Double. <% determines that it can process any object that
+    * extends from Iterable i.e. List, Vector, Seq, Queue, etc.
+    *
+    * @param targetTypeIterable an iterable that needs to be converted each element from TargetType to Double
+    * @return
+    */
+  def toDoubles[A <% Traversable[TargetType], B <% Traversable[Double]](targetTypeIterable: A): B
+
+  /**
+    * Converts a DataFrame column from Seq[Double] to a Seq[TargetType]. If the TargetType is Double, it will
```

Comment:
comment about what it will do in the case of Double should go with the Double implementation